ctest-next/templates/test.rs

```diff
@@ -82,7 +83,430 @@ mod generated_tests {
             for (i, (&b1, &b2)) in ptr1_bytes.iter().zip(ptr2_bytes.iter()).enumerate() {
                 // HACK: This may read uninitialized data! We do this because
                 // there isn't a good way to recursively iterate all fields.
-                check_same_hex(b1, b2, &format!("{{ ident }} value at byte {}", i));
+                check_same_hex(b1, b2, &format!("{{ ident }} value at byte {i}"));
+            }
+        }
+    }
+    {%- endif +%}
+    {%- endfor +%}
+
+    {%- for alias in ffi_items.aliases() +%}
+    {%- let ident = alias.ident() +%}
+
+    /// Test that the size and alignment of the aliased type is the same in both Rust and C.
+    /// 
+    /// This can fail if a different type is used on one side, and uses the built in size and
+    /// alignment functions to check.
+    pub fn size_align_{{ ident }}() {
+        extern "C" {
+            fn __test_size_{{ ident }}() -> u64;
+            fn __test_align_{{ ident }}() -> u64;
+        }
+        unsafe {
+            check_same(mem::size_of::<{{ ident }}>() as u64,
+                __test_size_{{ ident }}(), "{{ ident }} size");
+            check_same(mem::align_of::<{{ ident }}>() as u64,
+                __test_align_{{ ident }}(), "{{ ident }} align");
+        }
+    }
+
+    {%- if translator.has_sign(ffi_items, alias.ty) +%}
+
+    /// Test that the aliased type has the same sign (signed or unsigned) in both Rust and C.
+    ///
+    /// This check can be performed because `!(0 as _)` yields either -1 or the maximum value
+    /// depending on whether a signed or unsigned type is used. This is simply checked on both
+    /// Rust and C sides to see if they are equal.
+    pub fn sign_{{ ident }}() {
+        extern "C" {
+            fn __test_signed_{{ ident }}() -> u32;
+        }
+        unsafe {
+            check_same(((!(0 as {{ ident }})) < (0 as {{ ident }})) as u32,
+                    __test_signed_{{ ident }}(), "{{ ident }} signed");
+        }
+    }
+    {%- endif +%}
+
+    {%- if self::should_roundtrip(generator, ident) +%}
+
+    /// Generates a padding map for a specific type.
+    ///
+    /// Essentially, it returns a list of bytes, whose length is equal to the size of the type in
+    /// bytes. Each element corresponds to a byte and has two values. `1` if the byte is padding,
+    /// and `0` if the byte is not padding.
+    ///
+    /// For type aliases, the padding map is all zeroes.
+    fn roundtrip_padding_{{ ident }}() -> Vec<u8> {
+        vec![0; mem::size_of::<{{ ident }}>()]
```

Comment:
May as well use the prelude

```suggestion
        vec![0; size_of::<{{ ident }}>()]
```